modin/experimental/core/execution/native/implementations/hdk_on_native/partitioning/partition_manager.py

```diff
@@ -178,7 +178,13 @@ def _get_unsupported_cols(cls, obj):
 
             try:
                 at = pyarrow.Table.from_pandas(obj, preserve_index=False)
-            except (pyarrow.lib.ArrowTypeError, pyarrow.lib.ArrowInvalid) as err:
+            except (
+                pyarrow.lib.ArrowTypeError,
+                pyarrow.lib.ArrowInvalid,
+                ValueError,
+            ) as err:
+                if (type(obj) == ValueError) and obj.columns.is_unique:
+                    raise err
```

Comment:
```suggestion
            except ValueError:
                raise
            except (pyarrow.lib.ArrowTypeError, pyarrow.lib.ArrowInvalid) as err:
```

also I wonder how this relates to "support of duplicate columns", feels like it's fixing something unrelated